### PR TITLE
Feat/refactoring Getting started with foundry 

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: docs/02-developers/05-smart-contracts/05-foundry/deploy-smart-contracts.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+sidebar_label: Deploy Smart Contract
+sidebar_position: 105
+title: Deploy Smart Contract
+description: "Learn how to deploy your Rootstock smart contract using forge."
+tags: [guides, developers, smart contracts, rsk, rootstock, foundry, dApps]
+---
+
+In this section, you'll deploy `counter` smart contract to the Rootstock network using Foundry.
```
Comment (owans) [inline docs/02-developers/05-smart-contracts/05-foundry/deploy-smart-contracts.md]:
```suggestion
In this section, you'll deploy a `counter` smart contract to the Rootstock network using Foundry.
```

File: docs/02-developers/05-smart-contracts/05-foundry/deploy-smart-contracts.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Step 1: Deployment Script
+You will see a directory called `deploy` in the root of your project. This is where you will see/write your deployment scripts. The demo `counter.sol` comes with a deployment script `counter.s.sol`, which contains:
```
Comment (owans) [inline docs/02-developers/05-smart-contracts/05-foundry/deploy-smart-contracts.md]:
```suggestion
You will see a directory called `deploy` in the root of your project. This is where you can view/write your deployment scripts. The demo `counter.sol` comes with a deployment script `counter.s.sol`, which contains:
```

File: docs/02-developers/05-smart-contracts/05-foundry/test-smart-contracts.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+---
+sidebar_label: Test Smart Contract
+sidebar_position: 104
+title: Test Smart Contract
```
Comment (owans) [inline docs/02-developers/05-smart-contracts/05-foundry/test-smart-contracts.md]:
```suggestion
title: Testing Smart Contracts using Foundry
```

File: docs/02-developers/05-smart-contracts/05-foundry/test-smart-contracts.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+description: "Learn how to test your Rootstock smart contract"
```
Comment (owans) [inline docs/02-developers/05-smart-contracts/05-foundry/test-smart-contracts.md]:
```suggestion
description: "Learn how to test your Rootstock smart contract using Foundry"
```

File: docs/02-developers/05-smart-contracts/05-foundry/test-smart-contracts.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+tags: [guides, developers, smart contracts, rsk, rootstock, foundry, dApps, ethers]
+---
+
+In this section, you'll set up a smart contract test and test it using `forge`.
+
+### Step 1: Test Script
+You will see a directory called `test` in the root of your project. This is where you will see/write your tests. The demo `counter.sol` comes with a test script `counter.t.sol`, which contains:
```
Comment (owans) [inline docs/02-developers/05-smart-contracts/05-foundry/test-smart-contracts.md]:
```suggestion
You will see a directory called `test` in the root of your project. This is where you can view/write your tests. The demo `counter.sol` comes with a test script `counter.t.sol`, which contains:
```

File: docs/02-developers/05-smart-contracts/05-foundry/configure-foundry-rootstock.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+---
+
+### Environment Configuration
+
+Once you have an account with a private key, create a `.env` file in the root of the foundry project and add the variables. 
+
+Foundry automatically loads a `.env` file present in the project directory.
+
+The `.env` file should follow this format:
+
```
Comment (owans) [inline docs/02-developers/05-smart-contracts/05-foundry/configure-foundry-rootstock.md:16]: Add a note on how to get the RPC API URL in /developers/rpc-api/setup/

Comment (ghost) [reply docs/02-developers/05-smart-contracts/05-foundry/configure-foundry-rootstock.md:16]: okay noted, I will add this one asap

Comment (ghost) [reply docs/02-developers/05-smart-contracts/05-foundry/configure-foundry-rootstock.md:16]: I added this section @owans please check, thanks so much !!

**Review by owans — APPROVED** [review]
LGTM!

**Pull request merged**